pkg/registry/csv.go

```diff
@@ -412,3 +414,61 @@ func (csv *ClusterServiceVersion) GetSubstitutesFor() string {
 	}
 	return substitutesFor
 }
+
+func (csv *ClusterServiceVersion) UnmarshalJSON(data []byte) error {
+
+	if err := csv.UnmarshalSpec(data); err != nil {
+		return err
+	}
+	if err := csv.UnmarshalTypeMeta(data); err != nil {
+		return err
+	}
+	if err := csv.UnmarshalObjectMeta(data); err != nil {
+		return err
+	}
```

Comment:
It looks like this will always return on the first error. Something maybe worth considering for a better dx is having some form of error aggregation that can return all errors that occurred in unmarshalling. Non blocking though and could be done in a follow-up IMO

Reply:
In this case we specifically wanted to return on first error, since it means less output. :) 
But I can definitely agree that we could extend this to an aggregate and return all types in future.

Reply:
Note though that the original code using the default unmarshaler would have the same behavior where it would unmarshal only that which was present in the blob; a missing spec would still not be an error until we tried to use it later.  So this PR preserves the earlier behavior.

Reply:
Ah, good to know. Will mark this as resolved since it wasn't really meant to be an action item now but just something to consider for the future